nextcord/http.py

```diff
@@ -330,7 +333,16 @@ async def request(
                         data = await json_or_text(response)
 
                         # check if we have rate limit header information
-                        remaining = response.headers.get("X-Ratelimit-Remaining")
+                        is_global = bool(response.headers.get("X-RateLimit-Global", False))
+
+                        if is_global:
+                            limit = response.headers.get("X-RateLimit-Limit", "")
+                            remaining = response.headers.get("X-Ratelimit-Remaining", "")
+                        else:
+                            limit = response.headers["X-RateLimit-Limit"]
+                            remaining = response.headers["X-RateLimit-Remaining"]
```

Comment:
```suggestion
                        limit = response.headers.get("X-RateLimit-Limit", "")
                        remaining = response.headers.get("X-Ratelimit-Remaining", "")
```

As this is called for every request, and not all requests are rate limited. For example, logging in will KeyError